### PR TITLE
Restructure replacers

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
The markdown renderer

### Description
This PR refactors the component replacers with classes

### CODE REVIEW

File: src/components/editor/markdown-renderer/replace-components/quote-options/quote-options.tsx [hunk 1]
```diff
@@ -18,26 +17,26 @@ const findQuoteOptionsParent = (nodes: DomElement[]): DomElement | undefined =>
...
-  const childElements = paragraph.children || []
-  const optionsTag = childElements.find(isColorExtraElement)
-  if (!optionsTag) {
-    return
-  }
-  paragraph.children = childElements.filter(elem => !isColorExtraElement(elem))
-  const attributes = optionsTag.attribs
-  if (!attributes || !attributes['data-color']) {
-    return
+export class QuoteOptionsReplacer implements ComponentReplacer {
```
Comment (DerMolly) [inline src/components/editor/markdown-renderer/replace-components/quote-options/quote-options.tsx:20]: this file should then be called `quote-options-replacer.tsx`?

File: src/components/editor/markdown-renderer/replace-components/highlighted-fence/highlighted-fence.tsx [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
-import React, { ReactElement } from 'react'
+import React from 'react'
 import { HighlightedCode } from '../../../../common/highlighted-code/highlighted-code'
+import { ComponentReplacer, SubNodeConverter } from '../ComponentReplacer'
 
-const getElementReplacement = (codeNode: DomElement, index: number, counterMap: Map<string, number>): (ReactElement | undefined) => {
-  if (codeNode.name !== 'code' || !codeNode.attribs || !codeNode.attribs['data-highlight-language'] || !codeNode.children || !codeNode.children[0]) {
-    return
-  }
+export class HighlightedCodeReplacer implements ComponentReplacer {
```
Comment (DerMolly) [inline src/components/editor/markdown-renderer/replace-components/highlighted-fence/highlighted-fence.tsx:6]: this file should then be called `highlighted-fence-replacer.tsx`?

Comment (mrdrogdrog) [reply src/components/editor/markdown-renderer/replace-components/highlighted-fence/highlighted-fence.tsx]: renamed

File: src/components/editor/markdown-renderer/replace-components/quote-options/quote-options.tsx (hunk 1 above)
Comment (mrdrogdrog) [reply src/components/editor/markdown-renderer/replace-components/quote-options/quote-options.tsx]: renamed

**Pull request merged**